### DISCUSSION TITLE
or() using related tables

### DISCUSSION THREAD

**Discussion by juanantoniosantanamedina** [supabase/supabase#1021] [Questions]
Good evening!

I'm very new to supabase and I'm struggling when querying data.

This is my scenario:
Table users_medicals
   id
   fk_id_users
   fk_id_medical

Table medicals
   id
   name

There is a relation between users_medicals.fk_id_medical and medicals.id

I'm trying to query like:
supabase.from('users_medicals').select('*, medicals(name)').or('medicals.name.ilike.%something%')

I get this error:
failed to parse logic tree ((medicals.name.ilike.%some%))" (line 1, column 13)

What am I missing here. I've done it using fields from the table I'm querying but I guess I'm doing something wron with the related table field names.

Thanks in advance
Juan Antonio Santana

**Comment by steve-chavez** [comment]
Hey @juanantoniosantanamedina,

Try it like this:

```js
supabase.from('users_medicals').select('*, medicals(name)').or('name.ilike.%something%', { foreignTable: 'medicals'} )
```

Also check the docs for `or` on embedded tables here: https://supabase.io/docs/reference/javascript/or#use-or-on-foreign-tables

**Reply by juanantoniosantanamedina** [reply]
Hi Steve, thanks for your reply.
Is getting better, at least I don't have the error; but the result is not what I expected.

By doing that I get all the records from users_medicals but the property medicals, with the related record, only gets filled for the matching search.

It is like doing a left join and discarding the records from the right side of the join.

Thanks,
Juan Antonio Santana

**Reply by steve-chavez** [reply]
> It is like doing a left join

Ah, yes exactly. It's a left join. Making embedding use an inner join is being tracked here: https://github.com/PostgREST/postgrest/issues/1075

For now, the workaround is to can remove the null medicals client-side.
